CommunicationLayer/CommDeviceControl/UdpMessageForwarder.cpp

```diff
@@ -26,27 +26,69 @@
 #include <QByteArray>
 #include <QDebug>
 #include <QHostAddress>
+#include <QTextCodec>
+
 
 #include "InfrastructureLayer/Settings/I_Settings.h"
 #include "UdpMessageForwarder.h"
 
+
+namespace {
+    quint32 TIMEOUT = 120;
+    quint32 SLEEP_TIME = 2;
+}
+
 UdpMessageForwarder::UdpMessageForwarder(I_Settings& settings)
-: ipAddress_(settings.ipAddress())
-, port_(settings.udpPort())
 {
-    socket_.bind(ipAddress_, port_);
+    exchangeName_ = settings.exchangeName();
+    ipAddress_ = settings.ipAddress();
+    udpPort_ = settings.udpPort();
+    setupChannel();
 }
 
 UdpMessageForwarder::~UdpMessageForwarder()
 {
 }
 
+
+void UdpMessageForwarder::setupChannel()
+{
+    quint32 i = 0;
+    do {
+        if (i++) {
+            qWarning() << "UdpMessageForwarder: Attempting to reconnect";
+        }
+        try {
+            channel_ = AmqpClient::Channel::Create(ipAddress_.toStdString(), udpPort_);
+        } catch (std::exception&) {
+            if (channel_ == NULL) {
+                if (i == (TIMEOUT / SLEEP_TIME)) {
+                    qWarning() << "UdpMessageForwarder timed out waiting for connection to broker";
+                    throw;
+                }
+                qWarning() << "UdpMessageForwarder: error creating channel";
+                QThread::sleep(SLEEP_TIME);
+            }
+            else {
+                throw;
+            }
+        }
+
+
+    } while (channel_ == NULL);
+    channel_->DeclareExchange(exchangeName_.toStdString(), AmqpClient::Channel::EXCHANGE_TYPE_FANOUT);
+}
```

Comment:
I think the program should terminate. In our use case, we can't really turn the server on after the car turns on, so failing fast and having hermes erroring out makes sense I think.